File: my_first_package/src/hello_node.cpp
#include "rclcpp/rclcpp.hpp"

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);           // 初始化
    // 创建节点：cpp_hello_node
    auto node = rclcpp::Node::make_shared("cpp_hello_node");

    RCLCPP_INFO(node->get_logger(), "Hello World from C++!");

    rclcpp::shutdown();                 // 关闭
    return 0;
}

File: my_first_package/my_first_package/hello_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

def main(args=None):
    rclpy.init(args=args)                # 初始化 ROS 2
    node = Node("py_hello_node")         # 节点名：py_hello_node
    node.get_logger().info("Hello World from Python!") # 打印日志
    rclpy.shutdown()                     # 关闭

if __name__ == '__main__':
    main()